### ISSUE TITLE
Use custom URI scheme for connections

### ISSUE THREAD

**Issue by int19h** [Microsoft/RTVS#2981]
To avoid user confusion, we should adopt a custom scheme for our connection URIs, instead of using `http://` and `https://`.

There are a number of reasons for this. The most obvious one is that our default port number (5444) doesn't match that of either of those schemes, so we have to do funky autocorrection when editing connections. On top of that, we aren't actually connecting to arbitrary HTTP resources, but specifically to our server, talking our own protocol. While the protocol is implemented on top of HTTP, and that is a part of the contract (to ensure that things like proxies work), it's distinct enough to warrant its own scheme.

My suggestion would be `rrs://`, for R Remote Services (and to avoid tying it to RTVS, since it's more generic than that, and we intend to use it for other things in the future). This would also imply SSL.

To connect to non-SSL-secured servers, for debug/testing purposes only (which we only allow in Debug builds anyway), the counterpart would be `rrs-unsecured://`

The default port for that scheme would be 5444 if not specified. Host name and user information parts have the same meaning as they do in HTTP URIs. Path is allowed, but should always be `/` if present. Query string (`?...`) is not allowed. Fragment (`#...`) is allowed to indicate interpreter.

**Comment by int19h** [comment]
@MikhailArkhipov @jflam @smortaz, what do you think?

**Comment by AlexanderSher** [comment]
That would mean that we can't use `HttpClient`, correct?

**Comment by MikhailArkhipov** [comment]
I am not sure if this  simplifies user experience apart from typing different scheme. We already correct anyway so user doesn't have to type anything and I am not sure they generally care if port matches scheme.

**Comment by jflam** [comment]
I worry about the added complexity here, especially for administration scenarios. It's one more thing for us to explain to both users and administrators. Seems a small benefit that we would get for not having to explain to the user why they need to type :5444.

**Comment by int19h** [comment]
No, we'd still use `HttpClient`. We'd just produce an http:// URL from the one that user had input.

**Comment by int19h** [comment]
The benefits are many. First and foremost, it makes it clear that the URL is an RTVS remote URL, not something else. This also means that we can register a global URL handler for these, and then you could send people emails with a link, and they could add a connection to the server simply by clicking on it. It also makes it clear why we do what we do with the URLs when we sanitize them - i.e. why they have the magic port added, why paths are stripped etc.

Documentation-wise I think it would actually be easier, rather than more complex. Instead of having to explain everything that's "special" about our HTTP URLs, we simply document our scheme, and that's that.

Finally, it's simply the right thing to do. We have our own service and protocol. That it uses HTTP underneath is largely an implementation detail at this point. Exposing it to the users is more confusing than it is helpful.

**Comment by smortaz** [comment]
i haven't followed this in detail - but i'd say let's meet w carl & friends
1st about their remoting machinery, figure out what the combined story is
before we make further changes.

**Comment by int19h** [comment]
It is largely orthogonal to that, since it doesn't change anything about our architecture or implementation - it's only about what we present to the users as connection identifier.
